File: android/sdk/src/main/kotlin/world/humanlabs/quest/models/LaunchToken.kt
package world.humanlabs.quest.models

/**
 * Result of a server-side launch signing call (secure mode).
 *
 * In secure mode your backend signs the launch with the app secret and returns these
 * fields. The SDK combines them with the baked App ID + user ID to build the launch URL.
 * The app secret itself must **never** be embedded in the app — see `docs/SIGNING.md`.
 *
 * @property ts     Unix time (seconds or milliseconds) used in the signed message.
 * @property nonce  Random string, unique per launch (8–100 chars).
 * @property sig    Lowercase hex `HMAC-SHA256(app_secret, message)`.
 * @property locale Optional locale (`en` / `id` / `es` / `pt`) that was signed.
 */
data class LaunchToken(
    val ts: String,
    val nonce: String,
    val sig: String,
    val locale: String? = null,
)

File: ios/Sources/TheQuestOfferwall/Models/LaunchToken.swift
import Foundation

/// A short-lived, server-signed launch credential used in **secure mode**.
///
/// The values are produced by your backend (which holds the app secret) — see
/// `docs/SIGNING.md`. The SDK never signs launches itself; it only forwards the
/// signature you provide when building the offerwall launch URL.
public struct LaunchToken: Sendable {
    /// Unix timestamp (seconds or milliseconds) of when the signature was produced.
    /// Valid within ±5 minutes of server time.
    public let ts: String

    /// Random, per-launch nonce (8–100 characters).
    public let nonce: String

    /// Lowercase hex `HMAC-SHA256(app_secret, message)` signature.
    public let sig: String

    /// Optional locale (`en` / `id` / `es` / `pt`) that was included in the signed message.
    /// When present it overrides the device-derived locale so the signature stays valid.
    public let locale: String?

    /// Creates a launch token from values returned by your signing backend.
    public init(ts: String, nonce: String, sig: String, locale: String? = nil) {
        self.ts = ts
        self.nonce = nonce
        self.sig = sig
        self.locale = locale
    }
}
